### PR TITLE
New Feature -- bodc listing tests

### PR DESCRIPTION
**Pull request by marc-portier**
this introduces a new sparql template for listing all terms (skos-concepts) in a collection on the bodc vocab server

fixing a lot of oddities that were still lingering in the  __main__

shows that we should think about
- a decent test for the cli itself
- some further refactoring and cleaning up that code

### CODE REVIEW

File: pykg2tbl/__main__.py [hunk 1]
```diff
@@ -276,20 +273,18 @@ def main(sysargs=None):
             variables_template=vars_template, variables_given=params
         )
     query = template_service.build_sparql_query(
-        name=args.template_name, variables=params
+        name=args.template_name, **params
```
Comment (JotaFan) [inline pykg2tbl/__main__.py:276]: :+1:

File: tests/const.py [hunk 2]
```diff
@@ -20,3 +21,7 @@
     "https://raw.githubusercontent.com/"
     "ukgovld/registry-core/master/src/main/vocabs/registryVocab.ttl"
 )
+
+P06_DUMP_FILE = Path(__file__).parent / "data/20230605-P06-dump.ttl"
+
+FAKE_DUMP_FILE = Path(__file__).parent / "data/fake-dump.ttl"
```
Comment (JotaFan) [inline tests/const.py:27]: we are using os.path on some occasions to have the same logic.
I should only adopt one of the two cases I think. 
I personally have no preference whatsoever.

File: tests/test_bodc_queries.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        ttl_dump.exists()
+    ), f"need input file { str(ttl_dump) } for test to work"
+    in_memory: KGSource = KGSource.build(str(ttl_dump))
+    qry: str = DefaultQueryBuilder().build_sparql_query(
+        name="bodc-listing.sparql", cc="P06"
+    )
+
+    result: QueryResult = in_memory.query(sparql=qry)
+    assert result is not None, "there should be a result"
+    assert len(result) == 418, "the known dated dump had exactly 24 members"
```
Comment (JotaFan) [inline tests/test_bodc_queries.py:35]: Assertion not correct.
As in the message, the len(result) is known to be 24

File: tests/test_bodc_queries.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ttl_dump.exists()
+    ), f"need input file { str(ttl_dump) } for test to work"
+    in_memory: KGSource = KGSource.build(str(ttl_dump))
+    qry: str = DefaultQueryBuilder().build_sparql_query(
+        name="bodc-listing.sparql", cc="fake"
+    )
+
+    result: QueryResult = in_memory.query(sparql=qry)
+    assert result is not None, "there should be a result"
+    assert len(result) == 3, "the known fake dump has exactly 3 members"
```
Comment (JotaFan) [inline tests/test_bodc_queries.py:50]: Assertion not correct.
As in the message, the len(result) is known to be 3, but the test is failing

**Review by JotaFan — COMMENTED** [review]
Some tests failing.
Use of two different ways of handling paths.

But looks great to me. Not even  going to ask for changes. Although the tests need fixing otherwise the PR wont pass

File: tests/const.py (hunk 2 above)
Comment (marc-portier) [reply tests/const.py:27]: I prefer pathlib :)
it is limiting to python3 
but I don´t see that as a real limitation?

File: tests/test_bodc_queries.py (hunk 3 above)
Comment (marc-portier) [reply tests/test_bodc_queries.py:35]: have to check up - the 24 was actually ignoring a lot of matches that should be there - base on some optional part (vague memory)

I hope the same glitch is responsible for the 3->0 reduction on the 'fake' test

File: tests/const.py [hunk 5]
```diff
@@ -22,6 +20,5 @@
     "ukgovld/registry-core/master/src/main/vocabs/registryVocab.ttl"
 )
 
-P06_DUMP_FILE = Path(__file__).parent / "data/20230605-P06-dump.ttl"
-
-FAKE_DUMP_FILE = Path(__file__).parent / "data/fake-dump.ttl"
+P06_DUMP_FILE = SOURCES_PATH / "bodc/20230605-P06-dump.ttl"
```
Comment (JotaFan) [inline tests/const.py:23]: will this work in windows as well?

**Review by JotaFan — APPROVED** [review]

File: tests/test_bodc_queries.py (hunk 4 above)
Comment (marc-portier) [reply tests/test_bodc_queries.py:50]: data was not correct for the updated query

File: tests/test_bodc_queries.py (hunk 3 above)
Comment (marc-portier) [reply tests/test_bodc_queries.py]: query was updated to match better on the language of altLabel
and then also try avoid mulitple listings for the same member (with language variants in the altLabel)

correct result is 395

**Pull request merged**